sorting/tree_sort.cpp

```diff
@@ -0,0 +1,172 @@
+/**
+ * @file
+ * @brief Tree Sort Algorithm 
+ * More about it: https://en.wikipedia.org/wiki/Tree_sort
+ * @details
+ * Tree sort algorithm is a stable sorting algorithm that uses the property of how a BST stores data.
+ * Tree sort performs an inorder traversal on a BST to get the sorted order of elements.
+ * Worst case time-complexity of tree_sort is O(N*log(N)), given that the tree is balanced.
+ * To ensure that we use a balanced BST, we are using left-leaning Red-Black BST implementation of BST
+ * ensuring that our tree stays balanced all the time.
+ * Best case, Average case, the Worst case time complexity for tree_sort: N*log(N)
+ * Space complexity: O(n)
+ * @author [Puneet Tripathi](https://github.com/PuneetTri)
+ */
+
+#include<iostream>  // For standard I/O
+#include<cassert> // For assert
+#include<time.h> // For shuffling array using srand(time(NULL))
+
+// For the nodes of Binary Search Tree
+struct node {
+    int data;
+    node *left;
+    node *right;
+    bool RED; // Color of parent link
+};
+
+// Utility function to easily create new nodes
```

Comment:
namespaces missing. 😞

Reply:
@mishraabhinn sorry I don't understand what namespaces you're referring to. I've used std::cout for printing results, I might not be familiar with what you're asking. Can you suggest the required changes I need to make in my code

Reply:
![Screenshot 2022-10-24 at 14 30 56](https://user-images.githubusercontent.com/49574460/197489134-7d5de650-b8c3-4657-bc56-680ddc121b5f.png)

Reply:
![image](https://user-images.githubusercontent.com/43619937/197491596-c781eb12-4e43-46e8-9257-14c5fe8baaf1.png)
I've made changes and added namespaces, have I done it right? @mishraabhinn

Reply:
![image](https://user-images.githubusercontent.com/43619937/197491976-059330a0-80bf-44f8-bed4-b03f54b720ad.png)

Reply:
Looks good